src/js/workshop-comments.js

```diff
@@ -0,0 +1,144 @@
+		}
+		disableFields(false);
+	});
+
+	// Show failure message in case of error
+	XHR.addEventListener( "error", function() {
+		// Show failed comment submission message
+		submittedCommentMessage.setAttribute("style", "display: block;");
+		disableFields(false);
+	} );
+
+	// Set up our request
+	XHR.open( "POST", window.location.origin + "/api/comments" );
+
+	// The data sent is what the user provided in the form
+	const jsonFormData = JSON.stringify(Object.fromEntries(commentFormData));
+	XHR.setRequestHeader("Content-Type", "application/json");
+	XHR.send(jsonFormData);
+});
+
+/*
+ * Hide/show required and success messages
+ * @param {DOMElement} DOMselector - The DOM Element to parse.
+ * @param {Boolean} toDisplay - value is true to show and false to hide
+ * @param {String}[Optional] displayType - CSS display value to apply
+ */
+function applyVisibility(DOMselector, toShow, displayType = "none") {
+	toShow ? DOMselector.setAttribute("style", `display: ${displayType};`) : DOMselector.setAttribute("style", "display: none;");
+}
+
+/*
+ * Add new comment to the given workshop's page temporarily.
+ */
+function setSubmittedComment() {
+	// Add comment to the top of the comments section of workshop page
+	const newCommentDiv = document.createElement("div");
+	newCommentDiv.innerHTML = "<article class='comment submitted-comment'><p><span class='comment-name'></span> | <span class='comment-date'></span></p><p class='comment-text'></p></article>";
+	const commentsDiv = document.getElementsByClassName("comments")[0];
+	commentsDiv.prepend(newCommentDiv.firstChild);
+
+	const submittedCommentDiv = commentsDiv.firstChild;
+	const submittedDate = submittedCommentDiv.querySelector(".comment-date");
+	const submittedName = submittedCommentDiv.querySelector(".comment-name");
+	const submittedComment = submittedCommentDiv.querySelector(".comment-text");
+
+	const dateObject = new Date();
+	const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+
+	// Add values of comment
+	submittedDate.innerText = `${months[dateObject.getMonth()]} ${dateObject.getDate()}, ${dateObject.getFullYear()}`;
+	submittedName.innerText = nameDOMElement.value.trim();
+	submittedComment.innerText = commentDOMElement.value.trim();
+
+	// Show successful comment submission message
+	applyVisibility(document.querySelector(".post-success-message"), true , "block");
+}
+
+/*
+ * Disable the "post a comment" button and text fields when the post request is in the process of sending.
+ * @param {Boolean} sendingStatus - Boolean value indicating whether or not post request is in the send state.
+ */
+function disableFields(sendingStatus) {
+	// Comment form submit button
+	const postCommentButton = document.getElementById("post-comment");
+	sendingStatus ? postCommentButton.setAttribute("style", "background-color: #f3f3f3") : postCommentButton.setAttribute("style", "background-color: #fff");
```

Comment:
Same as the comment above, line 131 and 141 can be optimized to:

```
const backgroundColor = sendingStatus ? "#f3f3f3" : "#fff";
postCommentButton.setAttribute("style", "background-color: " + backgroundColor);
```

Similar change to line 141.